Makefile.am

```diff
@@ -246,8 +246,8 @@ BATS = \
 	test/functional/os-install/install-multiple.bats \
 	test/functional/os-install/install-no-fullfile-fallback.bats \
 	test/functional/os-install/install-no-packs.bats \
-	test/functional/os-install/install-statedir-cache.bats \
 	test/functional/os-install/install-statedir-cache-offline.bats \
+	test/functional/os-install/install-statedir-cache.bats \
```

Comment:
why did you swap these two tests? they were ordered correctly alphabetically

Reply:
The problem here is that sorting is more complicated than it looks. We need to define how we are going to consider non alphanumeric characters. There are different ways to sort that and I used the one defined by C, where '.' is located after '-'. 

I preferred that because the English sorting order, that is the one you're probably using, was creating an order that didn't look intuitive to me like:

    src/swupd_build_opts.h \
    src/swupd_build_variant.h \
    src/swupd_curl.h \
    src/swupd_curl_internal.h \
    src/swupd_exit_codes.h \
    src/swupd.h \
    src/swupd_internal.h \
    src/swupd_progress.c \
    src/swupd_progress.h \

placing swupd.h in the middle and not on top.

But I'm not picky on this. If you prefer to use English or any other LC_COLLATE I can change that

Reply:
well, is just that I think that when adding tests the easiest way to know the order (without thinking 😄 ) is just run an "ls -l" command in the directory of the new tests...

Reply:
@castulo Lol i had the same problem...I remember I wanted to add a test and didnt know which file to add it ..because didnt know which file had highest test no

Reply:
@otaviobp so to me it actually feels more intuitive to keep the alphabetical order because as an example, taking these picky tests (ordered alphabetically):
```
...
repair-picky.bats
repair-picky-downgrade.bats
repair-picky-ghosted.bats
repair-picky-ghosted-missing.bats
```
repair-picky.bats is placed on top of the repair-picky-<VARIANTS>, this makes sense because repair-picky.bats is probably the file with the most basic picky tests, since it has no variant written in the name, while the other files are obviously for more particular cases of picky....

Having said that if you want to keep this order I won't fight it either :P , but you'll have to explain me what files have precedence in this case so I know for future cases.

Reply:
> @castulo Lol i had the same problem...I remember I wanted to add a test and didnt know which file to add it ..because didnt know which file had highest test no

btw @karthikprabhuvinod you don't have to add a new test in the file that had the latest test ID, you need to add the new test in the file that makes more sense based on the theme or topic of the file. So in this example again if I was adding a new test that had to do with "--picky" and a ghosted I would probably add it in the repair-picky-ghosted.bats file, regardless of what was the last test ID in that file. You still need to use the next available ID when adding your test, but IDs don't have to be in order within those files. Makes sense?

Reply:
@castulo makes sense. thanks for clarifiying